src/main/java/com/prgrms/domain/user/UserController.java

```diff
@@ -22,18 +27,65 @@ public UserController(UserService service) {
         this.service = service;
     }
 
-    @GetMapping("/{id}")
-    public ResponseEntity<Response> getUserById(@PathVariable Long id) {
+    @GetMapping("/me")
+    public ResponseEntity<Response> getLoggedUserInfo(HttpServletRequest request)
+        throws AccessDeniedException {
 
-        Response userById = service.findUserById(id);
-        return ResponseEntity.ok(userById);
+        long loggedUserId = Long.parseLong(getCookieUserId(request).getValue());
+        Response userInfo = service.findUserById(loggedUserId);
+
+        return ResponseEntity.ok(userInfo);
+    }
+
+    @PostMapping()
+    public ResponseEntity<Void> userSignUp(@RequestBody Request userDto) {
+
+        service.insertUser(userDto);
+        URI loginPageUri = URI.create("api/users/login");
+
+        return ResponseEntity.created(loginPageUri)
+            .build();
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<Void> login(@RequestBody Login loginDto,
+        HttpServletResponse servletResponse) {
+
+        Response response = service.login(loginDto);
+        setCookieUserId(response.getUserId(), servletResponse);
+
+        URI myPageUri = URI.create("api/users/me");
+
+        return ResponseEntity.ok()
+            .location(myPageUri)
+            .build();
+    }
+
+    @PostMapping("/logout")
+    public ResponseEntity<Void> logout(HttpServletResponse response,HttpServletRequest request)
+        throws AccessDeniedException {
+
+        Cookie cookieUserId = getCookieUserId(request);
+        cookieUserId.setValue("");
+        response.addCookie(cookieUserId);
+
+        return ResponseEntity.ok()
+            .build();
+    }
+
+    private void setCookieUserId(Long userId, HttpServletResponse response) {
+
+        Cookie userCookie = new Cookie("userId", String.valueOf(userId));
+        response.addCookie(userCookie);
     }
 
-    @PostMapping
-    public ResponseEntity<Void> registerUser(@RequestBody Request userDto) {
+    private Cookie getCookieUserId(HttpServletRequest request) throws AccessDeniedException {
 
-        Response response = service.insertUser(userDto);
-        return ResponseEntity.created(URI.create("/users/" + response.getUserId())).build();
+        return Arrays.stream(request.getCookies())
+            .filter(i -> "userId".equals(i.getName()))
+            .filter(i -> !"".equals(i.getValue()))
```

Comment:
```suggestion
            .filter(cookie -> "userId".equals(cookie.getName()))
            .filter(cookie -> !"".equals(cookie.getValue()))
```

축약어를 사용하지말아주세요. 나중에 코드를 같이사용할텐데,  다른사람이 이해하기 어려울 수 있어요